llvm: passing byval structs as ptr byval(<ty>) intead of directly

kcbanner opened the pull request:
Experimental fix for https://github.com/ziglang/zig/issues/14355

Vexu commented:
This is not the correct approach, you should instead implement `ParamTypeIterator.nextInner` for `Stdcall`. It should be simpler than the `.C` version since you only need to handle `x86-windows`. You'll probably also want to look at other places where calling conventions are implemented like `firstParamSRet` and `lowerFnRetTy`.

kcbanner commented:
> This is not the correct approach, you should instead implement `ParamTypeIterator.nextInner` for `Stdcall`. It should be simpler than the `.C` version since you only need to handle `x86-windows`. You'll probably also want to look at other places where calling conventions are implemented like `firstParamSRet` and `lowerFnRetTy`.

@Vexu thanks for the guidance! 

I've implemented this using the approach you suggested, and added c_abi tests to cover this case.

kcbanner commented:
Thanks for the edits! Looks like there is an LLD crash in LTO when building for windows on linux.

```
2023-01-19T15:27:20.0499931Z error(link): /home/ci/actions-runner8/_work/zig/zig/build-debug/stage3-debug/bin/zig terminated with stderr:
2023-01-19T15:27:20.0500994Z Assertion failed: !GlobalRes.Prevailing && "Multiple prevailing defs are not allowed" (/home/andy/dev/zig-bootstrap/llvm/lib/LTO/LTO.cpp: addModuleToGlobalRes: 555)
```

Going to disable the windows c_abi tests when compiling on linux:
